samples/sensor/icp101xx/src/main.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <stdio.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ICP101XX_SAMPLE, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * Get a device structure from a devicetree node with compatible
+ * "invensense,icm101xx". (If there are multiple, just pick one.)
+ */
+static const struct device *get_icp101xx_device(void)
+{
+	const struct device *const dev = DEVICE_DT_GET_ANY(invensense_icp101xx);
+
+	if (dev == NULL) {
+		/* No such node, or the node does not have status "okay". */
+		printk("\nError: no device found.\n");
+		return NULL;
+	}
+
+	if (!device_is_ready(dev)) {
+		printk("\nError: Device \"%s\" is not ready; "
+		       "check the driver initialization logs for errors.\n",
+		       dev->name);
+		return NULL;
+	}
+
+	printk("Found device \"%s\", getting sensor data\n", dev->name);
+	return dev;
+}
+
+int main(void)
+{
+	const struct device *dev = get_icp101xx_device();
```

Comment:
const struct device *dev = DEVICE_DT_GET(DT_ALIAS(press_temp_altitude));

Any board overlay that supplies this devicetree alias then works with this sample. The sample could be renamed to something along those lines. BMP180 from bosch would also fit in this use case and sample, nothing icp101xxx specific should be needed.

Reply:
I understand the request, however there are already many Zephyr pressure sensor samples implemented this way (while not having any specific code), such as:
- BME280 from Bosh
- BME680 from Bosh
- DPS310 from Infineon
- LPS22HB from STMicroelectronics

Reply:
I think the point here is: does this sample really provide more than `sensor_shell` sample to an end-user in allowing them to easily exercise their sensor?

Arguably, the only added value is showing how one's DT overlay would look like but this is something that can also easily be done in the documentation of the binding directly if you really want to give users an example of how things could look like ; for example: https://github.com/zephyrproject-rtos/zephyr/blob/553a1c0063d81a6ab088f6632e5803b1e1d43418/dts/bindings/sensor/sensirion%2Cshtcx.yaml#L10-L21)

Reply:
I think the point here is: does this sample really provide more than `sensor_shell` sample to an end-user in allowing them to easily exercise their sensor?

Arguably, the only added value is showing how one's DT overlay would look like but this is something that can also easily be done in the documentation of the binding directly if you really want to give users an example of how things could look like ; for example: https://github.com/zephyrproject-rtos/zephyr/blob/553a1c0063d81a6ab088f6632e5803b1e1d43418/dts/bindings/sensor/sensirion%2Cshtcx.yaml#L10-L21)

Reply:
and to be clear, the "generic" samples you're referring to should probably just go away